### PR TITLE
fix: Attachment sent from slack doesn't reach to chatwoot

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://linear.app/chatwoot/issue/CW-2366/files-sent-from-slack-doesnt-reach-the-customer

https://www.loom.com/share/a606c17136284ec68e51a059508615e7

### CODE REVIEW

File: config/integration/apps.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ slack:
   id: slack
   logo: slack.png
   i18n_key: slack
-  action: https://slack.com/oauth/v2/authorize?scope=commands,chat:write,channels:read,channels:manage,channels:join,groups:read,groups:write,im:write,mpim:write,users:read,users:read.email,chat:write.customize,channels:history,groups:history,mpim:history,im:history
+  action: https://slack.com/oauth/v2/authorize?scope=commands,chat:write,channels:read,channels:manage,channels:join,groups:read,groups:write,im:write,mpim:write,users:read,users:read.email,chat:write.customize,channels:history,groups:history,mpim:history,im:history,files:read
```
Comment (sojan-official) [inline config/integration/apps.yml:16]: should we add https://api.slack.com/scopes/files:write permission as well ?

File: app/services/integrations/slack/incoming_message_builder.rb [hunk 2]
```diff
@@ -93,7 +97,18 @@ def private_note?
   def create_message
     return unless conversation
```
Comment (sojan-official) [inline app/services/integrations/slack/incoming_message_builder.rb:98]: There seems to be a code climate issue here. could refactor this method into smaller chunks

File: config/integration/apps.yml (hunk 1 above)
Comment (muhsin-k) [reply config/integration/apps.yml]: Yes.

File: app/services/integrations/slack/incoming_message_builder.rb (hunk 2 above)
Comment (muhsin-k) [reply app/services/integrations/slack/incoming_message_builder.rb]: Fixed.

**Comment by pranavrajs** [comment]
Please break this down to smaller PR fixing only the change that is required to fix it. Rest should be taken as a separate chore.

**Comment by muhsin-k** [comment]
> Please break this down to smaller PR fixing only the change that is required to fix it. Rest should be taken as a separate chore.

Created new PR for the fix, https://github.com/chatwoot/chatwoot/pull/7852, Other changes will address in a separate PR.

**Pull request closed without merging**